src/main/scala/chisel3/util/Math.scala

```diff
@@ -99,3 +98,34 @@ object isPow2 {
   def apply(in: BigInt): Boolean = in > 0 && ((in & (in-1)) == 0)
   def apply(in: Int): Boolean = apply(BigInt(in))
 }
+
+
+object unsignedBitLength {
+  /** Return the number of bits required to encode a specific value, assuming no sign bit is required.
+    *
+    * Basically, `n.bitLength` unless n is 0 (since Java believes 0.bitLength == 0), in which case the result is `1`.
+    * @param in - the number to be encoded.
+    * @return - an Int representing the number of bits to encode.
+    */
+  def apply(in: BigInt): Int = {
+    require(in >= 0)
+    if (in == 0) {
+      // 0.bitLength returns 0 - thank you Java.
+      1
+    } else {
+      in.bitLength
+    }
+  }
+}
+
+object signedBitLength {
+  /** Return the number of bits required to encode a specific value, assuming a sign bit is required.
+    *
+    * Basically, `n.bitLength` + 1.
+    * @param in - the number to be encoded.
+    * @return - an Int representing the number of bits to encode.
+    */
+  def apply(in: BigInt): Int = {
+    in.bitLength + 1
```

Comment:
Is this always true? With a signed int you can go one negative number lower than the max positive number right (for example, 16-bit signed is -32768 to 32767)?

Reply:
BigInt.bitLength explicitly assumes two's complement, so this is correct.

Reply:
Yep - just tried this in the repl and `BigInt(32767).bitLength`, `BigInt(32768).bitLength` both return 15